x-pack/plugins/ml/public/application/jobs/jobs_list/components/datafeed_modal/edit_query_delay.tsx

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useCallback, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { i18n } from '@kbn/i18n';
+import { cloneDeep } from 'lodash';
+import {
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiToolTip,
+} from '@elastic/eui';
+import { estypes } from '@elastic/elasticsearch';
+
+import { ml } from '../../../../services/ml_api_service';
+import { useToastNotificationService } from '../../../../services/toast_notification_service';
+import { DATAFEED_STATE } from '../../../../../../common/constants/states';
+import { Datafeed } from '../../../../../../common/types/anomaly_detection_jobs';
+
+const tooltipContent = i18n.translate(
+  'xpack.ml.jobsList.datafeedModal.editQueryDelay.tooltipContent',
+  {
+    defaultMessage: 'Cannot update query delay when datafeed is running.',
+  }
+);
+
+interface DatafeedWithStats extends Datafeed {
+  state?: estypes.DatafeedState;
+  timing_stats?: estypes.DatafeedTimingStats;
+}
+
+export const EditQueryDelay: FC<{
+  datafeedConfig: DatafeedWithStats;
```

Comment:
as mentioned in a previous comment, it would be better to only work with `Datafeed` objects if possible.